ClientSupport/src/main/java/io/deephaven/clientsupport/gotorow/SeekRow.java

```diff
@@ -45,155 +42,147 @@ public SeekRow(long startingRow, String columnName, Object seekValue, boolean in
+                            return true;
+
+                        } else {
+                            // current row is seek value, check next row
+                            if (compareResult == 0 && startingRow < rowSet.size() - 1) {
+                                final Comparable nextValue = (Comparable) columnSourceGet(rowSet.get(startingRow + 1));
+                                if (nullSafeCompare(nextValue, (Comparable) seekValue) == 0) {
+                                    result.setValue(startingRow + 1);
+                                    return true;
+                                }
+                                // next row is not the seek value, loop to start and find the first value
+                                // algorithm is the same as if seek value is above the current row
+                            } else if ((compareResult < 0 && order.get() == SortingOrder.Ascending)
+                                    || (compareResult > 0 && order.get() == SortingOrder.Descending)) {
+                                // current row is less than seek value and ascending
+                                // current row is greater than seek value and descending
+                                // which means seek value is below the current row, find the first occurrence
+                                result.setValue(
+                                        findEdgeOccurrence(rowSet, startingRow, rowSet.size() - 1, true,
+                                                order.get() == SortingOrder.Ascending));
+                                return true;
+                            }
+                            // seek value is above the current row
+                            // loop to start and find the first value
+                            result.setValue(findEdgeOccurrence(rowSet, 0, startingRow, true,
+                                    order.get() == SortingOrder.Ascending));
+                            return true;
+                        }
+                    }
 
-        if (isBackward) {
-            row = findRow(table, index, 0, (int) startingRow);
-            if (row >= 0) {
-                return row;
-            }
-            row = findRow(table, index, (int) startingRow, (int) index.size());
-            if (row >= 0) {
-                return row;
-            }
-        } else {
-            row = findRow(table, index, (int) startingRow + 1, (int) index.size());
-            if (row >= 0) {
-                return row;
-            }
-            row = findRow(table, index, 0, (int) startingRow + 1);
-            if (row >= 0) {
-                return row;
-            }
-        }
+                    long row;
+                    if (isBackward) {
+                        row = findRow(rowSet, 0, (int) startingRow);
+                        if (row >= 0) {
+                            result.setValue(row);
+                            log.info().append("found 1").endl();
```

Comment:
I'm guessing these log statements were for testing/debugging and can be removed?